### PR TITLE
Include TypeScript files in src to improve debugging from Backstage instance

### PR DESCRIPTION
**Pull request by taras**
## Hey, I just made a Pull Request!

This is PR is a follow up to [discussion in Discord](https://discord.com/channels/687207715902193673/687207715902193679/833040308006420570) and related to https://github.com/backstage/backstage/issues/2785.

The sourcemaps reference TypeScript files that are not included in distribution bundles. The result is that debugging from a Backstage instance is difficult because the debugger can't show the original code. In this PR, I changed every distributed packages to include `src` directory in package.json#files. This should cause src to be included with the bundles.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by freben** [comment]
Cool! Two things:

- This should also modify the create-app and cli create-plugin templates right? So that future creations also get this.

- As you do that, maybe break out a separate changeset for the create-app package, so that you can give some brief instructions there for how people can make the same addition to their local packages, in case they are interested in that. I guess it kind of depends right?

**Comment by taras** [comment]
@freben 

*create-app* - instance apps are usually not distributed. The package.json template doesn't currently have files entry - which I think default to including everything. 

[x] cli create-plugin templates - updated and created a separate changeset for it.

**Review by OrkoHunter — APPROVED** [review]
👍

**Comment by freben** [comment]
Ahh you are right, the template actually doesn't make any "library" packages of course.

**Comment by iciclespider** [comment]
Is that going to cause `isLocal` to be true here and result in a different behavior when used as a package? https://github.com/backstage/backstage/blob/master/packages/cli/bin/backstage-cli#L22
```
const isLocal = require('fs').existsSync(path.resolve(__dirname, '../src'));
```

**Comment by taras** [comment]
> Is that going to cause `isLocal` to be true here and result in a different behavior when used as a package? https://github.com/backstage/backstage/blob/master/packages/cli/bin/backstage-cli#L22
> 
> ```
> const isLocal = require('fs').existsSync(path.resolve(__dirname, '../src'));
> ```

@iciclespider Yes, I think so. I could remove dist from backstage-cli package or we could find another way to detect this.

**Review by Rugvip — COMMENTED** [review]
Thanks for looking into this! 👍

File: plugins/scaffolder/package.json [hunk 1]
```diff
@@ -74,6 +74,8 @@
     "msw": "^0.21.2"
   },
   "files": [
-    "dist"
+    "dist",
+    "src",
+    "src"
```
[the remark this answers is not in the document]
Comment (taras) [reply plugins/scaffolder/package.json:78]: Extra copypasta

**Comment by freben** [comment]
You forgot the DCO check.

File: packages/backend-common/package.json [hunk 2]
```diff
@@ -99,7 +99,8 @@
   },
   "files": [
     "dist",
-    "config.d.ts"
+    "config.d.ts",
+    "src"
```
[the remark this answers is not in the document]
Comment (taras) [reply packages/backend-common/package.json:103]: I spent some time trying to figure out what options we have here. It doesn't look good. The `files` property is not well documented. The best explanation that I could find is that it should be treated as `include`. It doesn't seem to support `!`, so `!src/**/*.test.*` doesn't match anything. I tried adding `.npmignore` but it didn't seem to change anything and it's risky ([according to](https://medium.com/@jdxcode/for-the-love-of-god-dont-use-npmignore-f93c08909d8d)). 

The best approach that I can come up with is either move test files into `__tests__` directory or ship them in the package.

**Review by Rugvip — COMMENTED** [review]
As part of the webpack 5 migration we've played around with sourcemaps a bit and realized that the ones we have in master are just broken, and we don't need to add `src/` to the published packages. The sourcemaps of the bundle output are supposed to contain the TypeScript source, but they don't, as they instead contain the post-transpile output, giving us something inbetween the TypeScript source and minified code.

The sourcemaps of the individual packages looks completely fine though, with proper relative source file references as well as inlined typescript code, we what we likely need to do is figure out how to fix sourcemaps in the webpack bundle.

Either way we want to avoid bundling the source again, as it's already included in the sourcemap files themselves.

**Comment by taras** [comment]
@Rugvip should I close this PR and try to figure out what's wrong with webpack?

**Comment by Rugvip** [comment]
@taras yeah I think so, as this isn't gonna be the fix we want since it bundles the source twice.

**Pull request closed without merging**